internal/v1/handler_blueprints.go

```diff
@@ -160,11 +167,20 @@ func (h *Handlers) CreateBlueprint(ctx echo.Context) error {
 	if blueprintRequest.Customizations.Users != nil {
 		for _, user := range *blueprintRequest.Customizations.Users {
 			// Make sure every user has either ssh key or password set
-			if user.Password == nil && user.SshKey == nil {
+			if (user.Password == nil || len(*user.Password) == 0) && user.SshKey == nil {
 				return ctx.JSON(http.StatusUnprocessableEntity, HTTPErrorList{
 					Errors: []HTTPError{{
 						Title:  "Invalid user",
-						Detail: "User must have either a password or an SSH key set.",
+						Detail: fmt.Sprintf("User ('%s') must have either a password or an SSH key set.", user.Name),
+					}},
+				})
+			}
+			if user.IsRedacted() {
+				// NOTE: we could silenty set password to empty if ssh key is available
```

Comment:
Not sure about this, I would prefer the error to not asume something, frontend can correctly validate this edge case (enforce non-empty ssh_key or password)